### ISSUE TITLE
video autoplay on AMP

### ISSUE THREAD

**Issue by SIbu1981** [ampproject/amphtml#32631]
hi we use amp-video-iframe to serve our customized jw player in our article pages. Starting from this use case are there any chance to build video autoplay and sticky on our pages? 

I have a look to some implementation in the our market and someone reached the objective but using the amp player component like https://amp.dev/documentation/components/amp-brightcove/) not the amp-video-iframe.

URL: https://sport.sky.it/calcio/serie-a/2021/02/12/napoli-juventus-pirlo-conferenza-video/amp

Thanks 
Silvio

**Comment by kristoferbaxter** [comment]
I've added @alanorozco to the issue, but autoplay requires ensuring the content is muted by default. I do not recall all of the details, but Alan can give a more in-depth answer to this question.

**Comment by alanorozco** [comment]
- **`autoplay`** works if you use `amp-video-iframe`. As long as you [follow the instructions](https://amp.dev/documentation/components/amp-video-iframe/#integration-inside-the-frame) to implement `mute()`, it should work just fine.

- By **sticky**, do you mean allowing the video to stick to an edge of the viewport? If so, using `amp-video-iframe` in conjunction with [`amp-video-docking`](https://amp.dev/documentation/components/amp-video-docking) should enable that behavior.

**Comment by gilbertococchi** [comment]
Hi believe @SIbu1981 point was on how Docking is possible with amp-video-iframe.

amp-video-docking allows docking only if the video has not autoplay but it seems like other AMP certified players allow this behaviour.

**Comment by alanorozco** [comment]
> amp-video-docking allows docking only if the video has not autoplay but it seems like other AMP certified players allow this behaviour.

This is incorrect. [amp-video-iframe works just fine with `dock`, here's an example.](https://playground.amp.dev/?format=websites&_gl=1*v51xdo*_ga*YW1wLWhjeGphNFd5d0llY2F3ano4M3ZoM0E.#share=PCFkb2N0eXBlIGh0bWw+CjxodG1sIOKaoSBsYW5nPSJlbiI+CjxoZWFkPgo8bWV0YSBjaGFyc2V0PSJ1dGYtOCI+CiAgPHRpdGxlPmFtcC12aWRlby1pZnJhbWU8L3RpdGxlPgogIDxsaW5rIHJlbD0iY2Fub25pY2FsIiBocmVmPSJodHRwczovL2FtcC5kZXYvZG9jdW1lbnRhdGlvbi9leGFtcGxlcy9jb21wb25lbnRzL2FtcC12aWRlby1pZnJhbWUvaW5kZXguaHRtbCI+CiAgPG1ldGEgbmFtZT0idmlld3BvcnQiIGNvbnRlbnQ9IndpZHRoPWRldmljZS13aWR0aCI+CiAgPHNjcmlwdCBhc3luYyBjdXN0b20tZWxlbWVudD0iYW1wLXZpZGVvLWlmcmFtZSIgc3JjPSJodHRwczovL2Nkbi5hbXBwcm9qZWN0Lm9yZy92MC9hbXAtdmlkZW8taWZyYW1lLTAuMS5qcyI+PC9zY3JpcHQ+CiAgPHNjcmlwdCBhc3luYyBjdXN0b20tZWxlbWVudD0iYW1wLXZpZGVvLWRvY2tpbmciIHNyYz0iaHR0cHM6Ly9jZG4uYW1wcHJvamVjdC5vcmcvdjAvYW1wLXZpZGVvLWRvY2tpbmctMC4xLmpzIj48L3NjcmlwdD4KICA8c3R5bGUgYW1wLWJvaWxlcnBsYXRlPmJvZHl7LXdlYmtpdC1hbmltYXRpb246LWFtcC1zdGFydCA4cyBzdGVwcygxLGVuZCkgMHMgMSBub3JtYWwgYm90aDstbW96LWFuaW1hdGlvbjotYW1wLXN0YXJ0IDhzIHN0ZXBzKDEsZW5kKSAwcyAxIG5vcm1hbCBib3RoOy1tcy1hbmltYXRpb246LWFtcC1zdGFydCA4cyBzdGVwcygxLGVuZCkgMHMgMSBub3JtYWwgYm90aDthbmltYXRpb246LWFtcC1zdGFydCA4cyBzdGVwcygxLGVuZCkgMHMgMSBub3JtYWwgYm90aH1ALXdlYmtpdC1rZXlmcmFtZXMgLWFtcC1zdGFydHtmcm9te3Zpc2liaWxpdHk6aGlkZGVufXRve3Zpc2liaWxpdHk6dmlzaWJsZX19QC1tb3ota2V5ZnJhbWVzIC1hbXAtc3RhcnR7ZnJvbXt2aXNpYmlsaXR5OmhpZGRlbn10b3t2aXNpYmlsaXR5OnZpc2libGV9fUAtbXMta2V5ZnJhbWVzIC1hbXAtc3RhcnR7ZnJvbXt2aXNpYmlsaXR5OmhpZGRlbn10b3t2aXNpYmlsaXR5OnZpc2libGV9fUAtby1rZXlmcmFtZXMgLWFtcC1zdGFydHtmcm9te3Zpc2liaWxpdHk6aGlkZGVufXRve3Zpc2liaWxpdHk6dmlzaWJsZX19QGtleWZyYW1lcyAtYW1wLXN0YXJ0e2Zyb217dmlzaWJpbGl0eTpoaWRkZW59dG97dmlzaWJpbGl0eTp2aXNpYmxlfX08L3N0eWxlPjxub3NjcmlwdD48c3R5bGUgYW1wLWJvaWxlcnBsYXRlPmJvZHl7LXdlYmtpdC1hbmltYXRpb246bm9uZTstbW96LWFuaW1hdGlvbjpub25lOy1tcy1hbmltYXRpb246bm9uZTthbmltYXRpb246bm9uZX08L3N0eWxlPjwvbm9zY3JpcHQ+CiAgPHNjcmlwdCBhc3luYyBzcmM9Imh0dHBzOi8vY2RuLmFtcHByb2plY3Qub3JnL3YwLmpzIj48L3NjcmlwdD4KICA8c3R5bGUgYW1wLWN1c3RvbT4KICAgIHAgewogICAgICBmb250LWZhbWlseTogc2Fucy1zZXJpZjsKICAgICAgZm9udC1zaXplOiAyMnB4OwogICAgICBmb250LXdlaWdodDogYm9sZDsKICAgICAgbWFyZ2luOiAyMHB4OwogICAgfQogIDwvc3R5bGU+CjwvaGVhZD4KPGJvZHk+CiAgPHA+CiAgICBDbGljayBvbiB2aWRlbyBpZiBpdCdzIGF1dG9wbGF5aW5nLgogIDwvcD4KICA8YW1wLXZpZGVvLWlmcmFtZSBpZD0ibXlWaWRlbyIgc3JjPSIvc3RhdGljL3NhbXBsZXMvZmlsZXMvYW1wLXZpZGVvLWlmcmFtZS12aWRlb2pzLmh0bWwiIHdpZHRoPSI3MjAiIGhlaWdodD0iNDA1IiBsYXlvdXQ9InJlc3BvbnNpdmUiIGF1dG9wbGF5IGRvY2s+CiAgICA8YW1wLWltZyBwbGFjZWhvbGRlciBzcmM9Ii9zdGF0aWMvc2FtcGxlcy9pbWcvYW1wLXZpZGVvLWlmcmFtZS1zYW1wbGUtcGxhY2Vob2xkZXIuanBnIiBsYXlvdXQ9ImZpbGwiPgogICAgPC9hbXAtaW1nPgogIDwvYW1wLXZpZGVvLWlmcmFtZT4KICA8ZGl2IHN0eWxlPSJoZWlnaHQ6IDUwMHZoIj48L2Rpdj4KPC9ib2R5PjwvaHRtbD4K)

What you may be experiencing is that autoplaying videos cannot be docked. Videos with `autoplay` need to be interacted with (clicked) before they can be docked, which is intentional.

**Comment by gilbertococchi** [comment]
Thanks Alan, sorry I meant to say the same, the limitation is expected on amp-video-iframe + amp-video-docking but other combo with amp-video-docking with other players are indeed allowing docking without user interaction with a video in autoplay.

**Comment by alanorozco** [comment]
@gilbertococchi One player seems to get around this limitation somehow, which it should not. Tracking separately #32706
